lua/completion/signature_help.lua

```diff
@@ -33,34 +32,35 @@ M.autoOpenSignatureHelp = function()
       local client = vim.lsp.get_client_by_id(client_id)
       local handler = client and client.handlers['textDocument/signatureHelp']
       if handler then
-          handler(err, method, result, client_id)
-          return
+        handler(err, method, result, client_id)
+        return
       end
       if not (result and result.signatures and result.signatures[1]) then
         return
       end
-      local lines = vim.lsp.util.convert_signature_help_to_markdown_lines(result, filetype)
+      local lines, hl = vim.lsp.util.convert_signature_help_to_markdown_lines(result, filetype)
+      lines = vim.lsp.util.trim_empty_lines(lines)
       if vim.tbl_isempty(lines) then
         return
       end
 
       -- if `lines` can be trimmed, it is modified in place
       local trimmed_lines_filetype = vim.lsp.util.try_trim_markdown_code_blocks(lines)
-	  local opts = {}
-	  if vim.g.completion_popup_border then
-	    opts.border = vim.g.completion_popup_border
-	  end
-      local bufnr, _ = vim.lsp.util.open_floating_preview(
-        -- TODO show popup when signatures is empty?
+      local opts = {}
+      if vim.g.completion_popup_border then
+        opts.border = vim.g.completion_popup_border
+      end
+      local fbuf, fwin = vim.lsp.util.open_floating_preview(
         vim.lsp.util.trim_empty_lines(lines),
```

Comment:
This trim is superfluous as it's been trimmed in (new) line 42 already, right?

Reply:
Woops! my bad. fixed now

Reply:
You're welcome and thanks for this, works great for me.